nodelet/src/loader.cpp

```diff
@@ -312,8 +312,14 @@ bool Loader::load(const std::string &name, const std::string& type, const ros::M
   ManagedNodelet* mn = new ManagedNodelet(p, impl_->callback_manager_.get());
   impl_->nodelets_.insert(const_cast<std::string&>(name), mn); // mn now owned by boost::ptr_map
   try {
+	mn->st_queue->disable();
+	mn->mt_queue->disable();
+
     p->init(name, remappings, my_argv, mn->st_queue.get(), mn->mt_queue.get());
-    /// @todo Can we delay processing the queues until Nodelet::onInit() returns?
+
```

Comment:
if `p->init()` throws .. are the queues disabled for ever, for the whole manager, i.e. all nodelets?

Reply:
Uh, interesting point. I think it is not a problem, since it is only deactivated for the `ManagedNodelet` Object.

Line 312 creates a new `ManagedNodelet` for this nodelet with its own queues.
This ManagedNodelet Object and its queues are removed in the catch block on line 329.

We are just blocking callbacks for the nodelet. All other nodelets can still receive callbacks.